client/www/components/dash/Sandbox.tsx

```diff
@@ -198,307 +380,508 @@ export function Sandbox({
-                  {
-                    diagnosticCodesToIgnore: [
-                      // top-level await without export
-                      1375,
-                    ],
-                  },
-                );
+              label="Delete Sandbox"
+              icon={<TrashIcon />}
+            />
+
+            <IconButton
+              disabled={!hasUnsavedWork}
+              onClick={() => {
+                saveCurrent(selectedSandbox.trim());
               }}
+              icon={<Save strokeWidth={1.5} width={18} />}
+              label={`Save "${selectedSandbox}" with current sandbox`}
+              variant="subtle"
             />
+          </>
+        )}
+        <IconButton
+          icon={<PlusIcon />}
+          label="Save As..."
+          onClick={() => {
+            saveCurrentDialog.onOpen();
+          }}
+          variant="subtle"
+        ></IconButton>
+      </div>
+    );
+  };
+
+  return (
+    <>
+      <Dialog {...saveCurrentDialog}>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            saveCurrent(newSaveName.trim());
+            saveCurrentDialog.onClose();
+            setSelectedSandbox(newSaveName);
+            setNewSaveName('');
+          }}
+        >
+          <div className="pb-2">
+            Enter a name to save the current sandbox as.
           </div>
-        </div>
-        <div className="flex flex-col border-b dark:border-b-neutral-700">
-          <div className="flex flex-col gap-1 border-b bg-gray-50 px-2 py-1 text-xs dark:border-b-neutral-700 dark:bg-neutral-800">
-            Context
-          </div>
-          <div className="flex items-center gap-2 px-2 py-1">
-            <Label className="text-xs font-normal">
-              Set{' '}
-              <code className="border bg-white px-2 dark:border-neutral-600 dark:bg-neutral-800">
-                auth.email
-              </code>
-            </Label>
-            <EmailInput
-              key={app.id}
-              db={db}
-              email={runAsUserEmail}
-              setEmail={setRunAsUserEmail}
-              onEnter={execRef.current}
-            />
+          <TextInput
+            autoFocus
+            className="mt-0"
+            label={<div className="pb-0 pt-2 font-[400]">Sandbox Name</div>}
+            value={newSaveName}
+            onChange={setNewSaveName}
+          />
+          <div className="flex justify-end pt-2">
+            <Button type="submit">Save</Button>
           </div>
-        </div>
+        </form>
+      </Dialog>
+      <div className="flex w-full justify-between border-b bg-white p-2 px-3 dark:border-b-neutral-600 dark:bg-neutral-800">
+        <PresetManager />
+        <div className="flex items-center gap-8">
+          <IconButton
+            variant="subtle"
+            label="Copy link to sandbox"
+            onClick={() => {
+              navigator.clipboard.writeText(window.location.toString());
```

Comment:
It would be safer to construct this url ourselves, since there could be some delay between your change to the sandbox and the URL updating.